Project-2/README.md

```diff
@@ -10,42 +10,261 @@ Tip #4: Work together as a team.
+  
+  while ∃r-.{a} means that for all xs, there is element a, which is r-inverse related to it.
+
 
 7. There is a delightfully helpful subreddit called "ELI5" which stands for something like "explain it like I'm 5" where users post conceptually challenging questions and other users attempt to provide explanations in simple, jargon-free, terms that presumably a 5 year-old could understand. Using this as a model, explain the _finite model property_. Be sure to provide a simple example and explain when the property might be important, and when it is not so important. 
 
+Ansewr: 
+Bob loves playing games, and he hopes there is a game that can be played with different results forever. 
+One day, his father gives him a game, called Color-card: there are a number of color cards in a box. Each card has a different color, but it is full colored on one side, while on the other side it is a blank sheet with the colored dot. Once a time, the player chooses any number of cards from the box, and then use any number of cards he chose to make combinations. For example, the first time, Bob chooses five color cards from the box, red, yellow, blue, green, brown, then he uses three cards, and put them in such a order: 1, red (front side), 2, blue (back side), 3, green (back side). That is one combination. The player continues playing until there is a combination that repeats. If the player cannot continue playing, he/she fails. 
+However, if the player fails at one time, he/she can choose more or/and different cards, then play the next time. The player wins if the cards he chooses can allow him to play forever. 
+Unfortunately, Bob cannot win the game, because no matter how many cards he chooses (the number of cards is not limitless), and how hard he tries, he cannot continue playing without a repeated combination occurs.  
+We say that the Color-card game has finite model property. 
+Finite model property is important because it can be used to design a decidable algorithm for the satisfiability of its concepts. 
+
 8. Following up on the preceding , explain the _tree model property_. Be sure to provide a simple example and explain when the property might be important, and when it is not so important. 
 
+At another day, Bob's father gives him a different game, called Dot-and-Line. 
+There is a big board, a number of tacks and thread. The games can be played with two people. When it is your turn, the other player choose a number of tacks and tacks them on the board - he/she can choose any place to locate a tack. Then you use thread to connect the tacks (any one tack needs to be connected to at least one other tack) - have all tacks to be connected. The connection is such that from one tack to any other tack through the thread, there is only one way to go. If the other player finds there is more than two ways between any two tacks in your work, you lose. If you does not lose, then it is the other player's turn. You two play in turn until one loses. 
```

Comment:
Same comment as above, love the story, connect to generalizable features of the tree model property.

Reply:
Dear John,

Thank you for the comments! I will add the generalizable features of both properties.

Bests,
Jieming